vars/rhsaErrata.groovy

```diff
@@ -0,0 +1,166 @@
+import groovy.json.JsonSlurper
+
+class CVE {
+    String name
+    String impact
+    String jiraNumber
+    String bzNumber
+    String problemDescription
+
+    static def impactOrdering = [Critical: '1', Important: '2', Moderate: '3', Low: '4']
+}
+
+/**
+ * Parse a Json that is resulted from getting CVE Jiras.
+ * Note that this json must contains summary and description fields.
+ *
+ * @param cveJson a Json that contains CVE Jiras
+ * @return a list of CVE
+ */
+def getCVEList(String cveJson) {
+    def resultMap = new JsonSlurper().parseText(cveJson)
+
+    def cveList = []
+    def jirasNumber = resultMap.issues.key
+    jirasNumber.eachWithIndex {jiraNumber, index ->
+        def summary = resultMap.issues.fields.summary[index]
+        def description = resultMap.issues.fields.description[index]
+
+        def cve = new CVE()
+        cve.name = getCVENumber(summary)
+        cve.impact = getCVEImpact(description)
+        cve.jiraNumber = jiraNumber
+        cve.bzNumber = getBZNumber(description)
+        cve.problemDescription = getProblemDescription(summary)
+        cveList.add(cve)
+    }
+    return cveList
+}
+
+/**
+ * Print a report containing the fields that are expected to be part of a RHSA errata.
+ * The report can be copied and pasted to the errata.
+ *
+ * @param cveList a list of CVE
+ */
+def printRHSAReport(List<CVE> cveList) {
+    println '===== RSHA Errata Report ====='
+
+    println "\n- Impact: ${getHigherCVEImpact(cveList)}"
+
+    println "\n- References: ${getReferenceLink(cveList)}"
+
+    println '\n- CVE Names:'
+    def cveNames = []
+    cveNames.addAll(cveList.name)
+    cveNames.sort()
+    println cveNames.join(' ')
+
+    println '\n- Bugs or JIRA Issues Fixed:'
+    def bugsAndJiras = []
+    bugsAndJiras.addAll(cveList.bzNumber)
+    bugsAndJiras.addAll(cveList.jiraNumber)
+    bugsAndJiras.sort()
+    println bugsAndJiras.join(' ')
+
+    println '\n- Problem Description:'
+    println 'Security Fix(es):\n'
+    println sortProblemDescriptions(cveList).join('\n\n')
+}
+
+/**
+ * Get the reference link used in the Errata.
+ * The link will always point to the higher CVE released as part of the Errata
+ *
+ * @param cveList a list of CVE
+ * @return the reference link
+ */
+def getReferenceLink(List<CVE> cveList) {
+    def link = 'https://access.redhat.com/security/updates/classification/#'
```

Comment:
shouldn't it be env agnostic? shouldn't it be part of the jenkins global env variables? wdyt?